pyiceberg/table/__init__.py

```diff
@@ -753,6 +807,143 @@ def update_table_metadata(base_metadata: TableMetadata, updates: Tuple[TableUpda
     return new_metadata.model_copy(deep=True)
 
 
+class InitialTableMetadataConstructor:
```

Comment:
I think we need a better way to construct an initial metadata from a chain of updates. The major challenge is that python does not allow an "empty" metadata like [Java](https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java#L332-L338).So we cannot directly use `_apply_table_update` or `update_table_metadata` here

Reply:
According to the Rest Catalog Open API spec, only the **format-version** and the **table-uuid** are the only required attributes of the entire Spec. What are your thoughts on updating the TableMetadataCommonFields Fields to align with the Rest Catalog Open API Spec? This will also allow us to create a Base TableMetadata, and use the existing **_apply_table_update** functions instead of having to write a separate constructor.